advance.cloudbuild.yaml

```diff
@@ -12,23 +12,23 @@
 steps:
 
 - id: 'Download dependencies'
-  name: golang:1.16.15
+  name: golang
```

Comment:
I tried to capture this in the PR description, but i think its reasonably safe to do this with go because of the care the language team puts into backward compatibility. That said, I understand if we'd prefer to pin to a minor version (e.g. `1.19`)

The issue for me is that this is redundant with the go.mod file, which (IIRC) renovate *does* update. so the robots will create problems that the humans will have to fix. 😿 

I'm not sure what's best here, and i'm open to alternate solutions.

Reply:
Two considerations:
* Go version support is current release -2 version. As a template showcasing how to create a new project, rather than a sample, focusing on the latest makes sense to me.
* Buildpack/Runtime support for the latest Go version could potentially lag.

By these considerations together, I think testing against latest stable is fine but we should consider if we prefer to have slightly lagging update of the go.mod version.